subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaCompile.java

```diff
@@ -87,7 +85,6 @@ public FileCollection getScalaCompilerPlugins() {
      * @param scalaCompilerPlugins Collection of Scala compiler plugins.
      * @since 6.4
      */
-    @Incubating
     public void setScalaCompilerPlugins(FileCollection scalaCompilerPlugins) {
```

Comment:
I think having this method added in 6.x a smell. Instead  `getScalaCompilerPlugins()` should return a ConfigurableFileCollection. I thought about changing this for 7.0, but it's been like this for some month now and it's probably not very important in this case.
If we ever get around to streamline this everywhere, we can deprecate this along with all the other places where we do things like this.